From: uazo <uazo@users.noreply.github.com>
Date: Sat, 9 Jul 2022 06:59:18 +0000
Subject: Remove window name on cross origin navigation

See also: https://trac.webkit.org/changeset/209076/webkit

Original License: GPL-2.0-or-later - https://spdx.org/licenses/GPL-2.0-or-later.html
License: GPL-3.0-only - https://spdx.org/licenses/GPL-3.0-only.html
---
 third_party/blink/renderer/core/loader/document_loader.cc | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/third_party/blink/renderer/core/loader/document_loader.cc b/third_party/blink/renderer/core/loader/document_loader.cc
--- a/third_party/blink/renderer/core/loader/document_loader.cc
+++ b/third_party/blink/renderer/core/loader/document_loader.cc
@@ -2529,7 +2529,7 @@ void DocumentLoader::CommitNavigation() {
     // that the name would be nulled and if the name is accessed after we will
     // fire a UseCounter. If we decide to move forward with this change, we'd
     // actually clean the name here.
-    // frame_->tree().setName(g_null_atom);
+    frame_->Tree().SetName(g_null_atom);
     frame_->Tree().ExperimentalSetNulledName();
   }
 
@@ -2540,6 +2540,7 @@ void DocumentLoader::CommitNavigation() {
     // TODO(shuuran): CrossSiteCrossBrowsingContextGroupSetNulledName will just
     // record the fact that the name would be nulled and if the name is accessed
     // after we will fire a UseCounter.
+    frame_->Tree().SetName(g_null_atom);
     frame_->Tree().CrossSiteCrossBrowsingContextGroupSetNulledName();
   }
 
--
2.25.1
